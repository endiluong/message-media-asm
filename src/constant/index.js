export const IMAGES = {
  LOAD: "IMAGES_LOAD",
  LOAD_SUCCESS: "IMAGES_LOAD_SUCCESS",
  LOAD_FAIL: "IMAGES_LOAD_FAIL",
  SET_OFFSET: "SET_OFFSET",
};
export const STATS = {
  LOAD: "STATS_LOAD",
  LOAD_SUCCESS: "STATS_LOAD_SUCCESS",
  LOAD_FAIL: "STATS_LOAD_FAIL",
};
export const API_KEY = "mDxBZ8g8muJniGkZE9PMb8JYc7Azs9yQ";
export const API_URL = "https://api.giphy.com";
export const API_PREFIX = "v1/gifs";
export const API_GENERAL_HEADERS = {
  Accept: "application/json",
  "Content-Type": "application/json",
};
export const API_BASE_URL = `${API_URL}/${API_PREFIX}`;

export const GIPHY_DATA_OFFSET = 20;
export const GIPHY_DATA_LIMIT = 20;
export const GIPHY_IMAGE_WEBP = "preview_webp";
export const GIPHY_IMAGE_JPG = "480w_still";
export const GIPHY_IMAGE_FULL = "downsized";
