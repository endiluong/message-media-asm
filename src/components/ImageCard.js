import React, { useState } from "react";
import { GIPHY_IMAGE_WEBP, GIPHY_IMAGE_JPG } from "../constant";
import LightBox from "./LightBox";

import "./ImageCard.scss";

const ImageCard = (props) => {
  const [isOpenLightBox, setOpenLightBox] = useState(false);

  const renderer = () => {
    const { image } = props;
    const { title, images, user } = image;
    return (
      <div className="col-6 col-md-4 col-lg-3">
        <section className="card">
          <div className="card-wrapper">
            <picture
              className="card-image"
              onClick={() => setOpenLightBox(true)}
            >
              <source srcSet={images[GIPHY_IMAGE_WEBP].url} type="image/webp" />
              <source srcSet={images[GIPHY_IMAGE_JPG].url} type="image/jpeg" />
              <img src={images[GIPHY_IMAGE_JPG].url} alt={title} />
            </picture>
            <div className="card-bottom d-flex align-items-center justify-content-between">
              <div className="mr-auto">
                <i className="fas fa-paperclip f6"></i>
              </div>
              <div className="ml-auto">
                <span className="ml-2">
                  <i className="fas fa-eye mr-2"></i>
                </span>
                <span className="ml-2">
                  <i className="fas fa-comment mr-2"></i>{" "}
                </span>
                <span className="ml-2">
                  <i className="fas fa-heart mr-2"></i>
                </span>
              </div>
            </div>
          </div>
          {user ? (
            <div className="card-user">
              <span className="card-user__avatar">
                <img src={user.avatar_url} alt={user.display_name} />
              </span>
              <span className="card-user__name">{user.username}</span>
            </div>
          ) : null}
          <LightBox
            isOpen={isOpenLightBox}
            onClose={() => {
              setOpenLightBox(false);
            }}
            {...image}
          />
        </section>
      </div>
    );
  };
  return renderer();
};

export default ImageCard;
