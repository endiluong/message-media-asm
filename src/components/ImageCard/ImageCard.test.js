import { shallow } from "enzyme";
import "../../setupTest";
import ImageCard from "./index";
import { GIPHY_IMAGE_WEBP, GIPHY_IMAGE_JPG } from "../../constant";

describe("Image Card", () => {
  let wrapper;

  beforeAll(() => {
    const props = {
      image: {
        id: "KfSUEbdncm6k8gCIBr",
        images: {
          [GIPHY_IMAGE_WEBP]: {
            height: "86",
            size: "32402",
            url:
              "https://media0.giphy.com/media/KfSUEbdncm6k8gCIBr/giphy-preview.webp?cid=59a25e7209btwcvu73t0cc9jgppok8n46ux9dfpsdd1m2lxx&rid=giphy-preview.webp",
            width: "152",
          },
          [GIPHY_IMAGE_JPG]: {
            height: "270",
            size: "1045466",
            url:
              "https://media0.giphy.com/media/KfSUEbdncm6k8gCIBr/480w_s.jpg?cid=59a25e7209btwcvu73t0cc9jgppok8n46ux9dfpsdd1m2lxx&rid=480w_s.jpg",
            width: "480",
          },
        },

        import_datetime: "2020-05-26 17:35:49",
        is_sticker: 0,
        rating: "g",
        slug: "jeopardy-football-alex-trebek-KfSUEbdncm6k8gCIBr",
        source: "",
        source_post_url: "",
        source_tld: "",
        title: "Alex Trebek Football GIF by Jeopardy!",
        trending_datetime: "2021-01-09 20:08:05",
        type: "gif",
        url:
          "https://giphy.com/gifs/jeopardy-football-alex-trebek-KfSUEbdncm6k8gCIBr",
        username: "jeopardy",
      },
    };
    wrapper = shallow(<ImageCard {...props} />);
  });
  it("should be render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should show lightbox on click", () => {
    wrapper.find("picture").simulate("click");
    const lightbox = wrapper.find("LightBox");
    expect(lightbox.prop("isOpen")).toBe(true);
  });
  it("should still render avatar with empty space when having no user", () => {
    expect(wrapper.find(".card-user").prop("src")).toBe(undefined);
    expect(wrapper.find(".card-user")).toHaveLength(1);
  });
});
