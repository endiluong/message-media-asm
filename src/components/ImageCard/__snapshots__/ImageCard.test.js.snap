// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Card should be render correctly 1`] = `ShallowWrapper {}`;
